docs/usage/jssprovider.md

```diff
@@ -0,0 +1,205 @@
+| `jss.fips`                         | `ivs.fipsMode`                      |
+| `jss.ocsp.enabled`                 | `ivs.ocspCheckingEnabled`           |
+| `jss.ocsp.policy`                  | `CryptoManager.setOCSPPolicy`       |
+| `jss.ocsp.responder.cert_nickname` | `ivs.ocspResponderCertNickname`     |
+| `jss.ocsp.responder.url`           | `ivs.ocspResponderURL`              |
+| `jss.password`                     | `CryptoManager.setPasswordCallback` |
+| `nss.config_dir`                   | `ivs.configDir`                     |
+| `nss.cert_prefix`                  | `ivs.certPrefix`                    |
+| `nss.cooperate`                    | `ivs.cooperate`                     |
+| `nss.force_open`                   | `ivs.forceOpen`                     |
+| `nss.java_only`                    | `ivs.javaOnly`                      |
+| `nss.key_prefix`                   | `ivs.keyPrefix`                     |
+| `nss.no_cert_db`                   | `ivs.noCertDB`                      |
+| `nss.no_mod_db`                    | `ivs.noModDB`                       |
+| `nss.no_pk11_finalize`             | `ivs.noPK11Finalize`                |
+| `nss.no_root_init`                 | `ivs.noRootInit`                    |
+| `nss.optimizeSpace`                | `ivs.optimizeSpace`                 |
+| `nss.pkix_verify`                  | `ivs.pkixVerify`                    |
+| `nss.pk11_reload`                  | `ivs.PK11Reload`                    |
+| `nss.pk11_thread_safe`             | `ivs.PK11ThreadSafe`                |
+| `nss.read_only`                    | `ivs.readOnly`                      |
+| `nss.secmod_name`                  | `ivs.secmodName`                    |
+
+Note that the parameters `installJSSProvider`, `removeSunProvider`, and
+`installJSSProviderFirst` are ignored, as they can be controlled by directly
+manipulating the `java.security` file.
+
+## Upgrading Old Code
+
+There are two paths to upgrade an older code base to a newer JSS version,
+using the JSSProvider interface:
+
+ 1. Continue using `CryptoManager.initialize(...)` as before. This gives the
+    local application the most control over the NSS DB path.
+ 2. Switch to using `java.security`-based configuration (either via local
+    policy with `-Djava.security.properties=/path` specified on the JVM
+    command line or via system-wide policy by modifying the
+    `$JAVA_HOME/conf/security/java.security` file). You can then remove the
+    call to `CryptoManager.initialize(...)`. If this call is necessary for
+    backwards-compatibility reasons (to support multiple JSS versions), it
+    would be sufficient to check the value of `CryptoManager.getInstance(...)`
+    before configuration:
+
+    ```java
+    try {
+        cm = CryptoManager.getInstance();
+    } catch (NotInitializedException nie) {
+        CryptoManager.initialize(...);
+        cm = CryptoManager.getInstance();
+    }
```

Comment:
Not sure it's necessary to tell people to change their code this way.
Theoretically people can continue to use the old API which will work
with the old and the new JSS:
```
CryptoManager.initialize(...);
CryptoManager cm = CryptoManager.getInstance();
```
Once they no longer need to support the old JSS they can simply drop
`CryptoManager.initialize()` and switch to `java.security`.

Reply:
This is the section under `java.security`. Presumably, if they're using this code snippet, they'd prefer a `java.security` JSS instance over a `CryptoManager` instance. This describe a fallback when either JSS doesn't support `java.security` or JSS wasn't initialized that way. 

I've added a comment saying they could throw this exception instead of catching it.

I don't think its reasonable to assume everyone is going to switch to `java.security` right away, and those that will, won't necessarily support it consistently.